### PR TITLE
Nix fixes for brand new nix users

### PR DESCRIPTION
**Pull request by ahobson**
## Summary

* Always set the `NIX_PROFILE` as our prior check would fail for new users
* Don't try to use bash4+ features when checking for `NIX_PROFILE`

## Verification Steps for the Author

I tested this in a fresh VM and it seems to fix the problem for me

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -288,13 +288,6 @@ export GOTENBERG_PORT=2000
 # interactions
 export DTOD_USE_MOCK=true
 
-# Client build flags
-#
-# Send error logs to the console for local development. Set to 'otel'
-# to send to the backend
-export REACT_APP_ERROR_LOGGING=console
```
Comment (melissathai) [inline .envrc:295]: Why is this being removed?

File: .envrc [hunk 2]
```diff
@@ -311,11 +304,7 @@ if [ ! -r .nix-disable  ] && has nix-env; then
   # set NIX_PROFILE so nix-env operations don't need to manually
   # specify the profile path
   #
-  if [ -w "/nix/var/nix/profiles/per-user/${LOGNAME}" ]; then
-    export NIX_PROFILE="/nix/var/nix/profiles/per-user/${LOGNAME}/mymove"
-  else
-    log_error "Cannot determine writable location for your NIX_PROFILE"
-  fi
+  export NIX_PROFILE="/nix/var/nix/profiles/per-user/${LOGNAME}/mymove"
```
Comment (melissathai) [inline .envrc:307]: 💯

File: .envrc (hunk 1 above)
Comment (ahobson) [reply .envrc]: Thank you! That was a copy error on my part

File: nix/update.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 set -euo pipefail
 
-if [ ! -v NIX_PROFILE ]; then
+if [ -z "${NIX_PROFILE:-}" ]; then
   echo "NIX_PROFILE not set, not installing globally"
```
Comment (rogeruiz) [inline nix/update.sh:6]: Is this a reason why we had `Bash +4` in our project?

Comment (ahobson) [reply nix/update.sh:6]: > Is this a reason why we had `Bash +4` in our project?

Yeah. MacOS doesn't ship with a modern bash, and trying to get all the right dependencies and PATH in place seems to be too brittle.

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈

This looks good to me in terms of what I am reading here. This looks like we should have better onboarding for folks using Nix. I think we should also port this back over into our Engineering Playbook as well so we can share this wider than just in the `transcom/mymove` repository.

**Pull request merged**